lib/IRGen/IRGenModule.cpp

```diff
@@ -1009,6 +1009,15 @@ llvm::SmallString<32> getTargetDependentLibraryOption(const llvm::Triple &T,
 void IRGenModule::addLinkLibrary(const LinkLibrary &linkLib) {
   llvm::LLVMContext &ctx = Module.getContext();
 
+  // Don't emit the FORCE_LOAD symbols and metadata when the compiler is 
+  // running on behalf of the debugger.  The debugger will read the LinkLibrary's
+  // from all the modules it sees, and hand load all the required dependencies.
+  // so it doesn't need this information.  And since the FORCE_LOAD symbol is
+  // weak it doesn't even tell us whether a required dependency is missing.
+  // So it serves no purpose in this case.
```

Comment:
Looks good, but the comment is no longer accurate. I think it can just be "The debugger can get autolink information directly from the LinkLibraries of the module, so there's no reason to generate it in the LLVM IR."